client/engine/engine.go

```diff
@@ -721,3 +734,12 @@ func (e *Engine) recordMessageMetrics(message protocols.Message) {
 	e.metrics.RecordQueueLength(fmt.Sprintf("msg_payload_size,sender=%s,receiver=%s", e.store.GetAddress(), message.To), totalPayloadsSize)
 	e.metrics.RecordQueueLength(fmt.Sprintf("msg_size,sender=%s,receiver=%s", e.store.GetAddress(), message.To), len(raw))
 }
+
+func (e *Engine) CompletedObjectiveChan(id protocols.ObjectiveId) <-chan struct{} {
```

Comment:
Instead of returning an empty struct should we return something with the objective id? While not strictly necessary it could be nice for debugging purposes (and maybe developer ergonomics?)

Reply:
I'm not sure that would help a lot... for developer ergonomics:

```go
<-CompletedObjectiveChan(id)
```
versus

```go
sameId := <-CompletedObjectiveChan(id)
```

Maybe for debugging but I don't see that yet. 

One option would be to return an enum for "rejected, failed, succeeded" for each id. 

The thing with this pattern is it does not work for repeated events, so we couldn't use it easily to do "objective updated" event emission multiple times. For that we would want to use a condition variable [`sync.Cond`.](https://pkg.go.dev/sync#Cond)